src/PostRector/Rector/UnusedImportRemovingPostRector.php

```diff
@@ -163,14 +164,19 @@ private function resolveUsedPhpAndDocNames(Namespace_|FileWithoutNamespace $name
     }
 
     /**
-     * @param string[]  $names
+     * @param string[] $names
      */
-    private function isUseImportUsed(UseUse $useUse, array $names, ?Name $namespaceName): bool
+    private function isUseImportUsed(UseUse $useUse, bool $isCaseSensitive, array $names, ?Name $namespaceName): bool
     {
         $comparedName = $useUse->alias instanceof Identifier
             ? $useUse->alias->toString()
             : $useUse->name->toString();
 
+        if (! $isCaseSensitive) {
+            $comparedName = strtolower($comparedName);
+            $names = array_map(strtolower(...), $names);
```

Comment:
this seems cause double iteration, to avoid performance regression for many names, I think it can be moved to loop at line 199+:

```php
foreach ($names as $name) {
    if (! $isCaseSensitive) {
        $name = strtolower($name);
    }

   // ...
}

Reply:
I wanted to think about it in the next PR, but I added additional optimization to this rule.

- lazy loading of lowercase names
- leaving the loop when no names are found. No names -> no usage.